### PR TITLE
enabled support for RESTAPI monitors

### PR DESCRIPTION
**Pull request by AlKapkone**
added support of REST API monitor type

### CODE REVIEW

**Comment by svenstaro** [comment]
Did you test this against a real setup?

**Comment by AlKapkone** [comment]
Tested locally and it's working as expected.
```
site24x7_monitor_latency_seconds{location="Amsterdam - NL",monitor_group="",monitor_name="example-api",monitor_type="RESTAPI"} 0.321
site24x7_monitor_latency_seconds{location="Amsterdam - NL",monitor_group="",monitor_name="example-front",monitor_type="URL"} 0.025
```

**Comment by AlKapkone** [comment]
Unfortunately, the command `curl https://accounts.zoho.eu/oauth/v2/token -X POST @curl-secrets` didn't work, so I used Postman for this purpose.

**Comment by AlKapkone** [comment]
Postman usage:
Screenshot 2024-04-05 at 16 57 30

**Comment by AlKapkone** [comment]
`cargo run -- --site24x7-endpoint site24x7.eu
    Finished dev [unoptimized + debuginfo] target(s) in 0.22s
     Running `target/debug/site24x7_exporter --site24x7-endpoint site24x7.eu`
14:02:20 [INFO] site24x7_exporter 1.0.1
14:02:20 [INFO] Using site24x7 endpoint: https://www.site24x7.eu/api
14:02:20 [INFO] Using Zoho endpoint: https://accounts.zoho.eu
14:02:20 [INFO] Not using any proxies
14:02:20 [INFO] Requesting access token from https://accounts.zoho.eu/oauth/v2/token
14:02:21 [INFO] Successfully acquired access token
14:02:26 [INFO] Serving metrics`

**Comment by svenstaro** [comment]
Could you rebase onto main please? That'll make the CI work.

**Comment by svenstaro** [comment]
Also please update the README and CHANGELOG.

**Comment by AlKapkone** [comment]
> Could you rebase onto master please? That'll make the CI work.
Should I merge my feature branch into the main branch of my fork?

**Comment by svenstaro** [comment]
You should fetch upstream/origin and rebase your feature branch on top of that so that the CI may run.

**Comment by AlKapkone** [comment]
I got it, thanks, done. I missed that you made some fresh changes.

**Comment by svenstaro** [comment]
Also update the features section of the README.

**Comment by svenstaro** [comment]
And it would be great if you could rebase this down into one commit.

File: README.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ This exporter currently supports these monitor types:
 - URL ["Website"](https://www.site24x7.com/help/admin/adding-a-monitor/website-monitoring.html)
 - HOMEPAGE ["Web Page Speed (Browser)"](https://www.site24x7.com/help/admin/adding-a-monitor/web-page-analyzer.html)
 - REALBROWSER ["Web Transaction (Browser)"](https://www.site24x7.com/help/admin/adding-a-monitor/webapplication-monitoring-realbrowser.html)
+- RESTAPI ["REST API"](https://www.site24x7.com/help/admin/adding-a-monitor/rest-api-monitor.html)
```
Comment (svenstaro) [inline README.md:18]: Put this in the same location as the enum variant so it's easy to spot the mapping at a glance.

Comment (svenstaro) [inline README.md:18]:
```suggestion
- RESTAPI ["REST API"](https://www.site24x7.com/help/admin/adding-a-monitor/rest-api-monitor.html)
- REALBROWSER ["Web Transaction (Browser)"](https://www.site24x7.com/help/admin/adding-a-monitor/webapplication-monitoring-realbrowser.html)
```

**Comment by AlKapkone** [comment]
Moved to https://github.com/svenstaro/site24x7_exporter/pull/342

**Pull request closed without merging**